tests/pytests/test_cursors.py

```diff
@@ -371,3 +371,25 @@ def testCursorDepletionNonStrictTimeoutPolicy(env):
         n_recieved += len(res) - 1
 
     env.assertEqual(n_recieved, num_docs)
+
+def test_timeoutError():
+    """Tests that the cursor returns a timeout error (accompanied by a `0`
+    cursor-id) in case of a timeout, when the timeout policy is
+    `ON_TIMEOUT FAIL`"""
```

Comment:
No zero cursor-id is tested here - Need the `ON_TIMEOUT RETURN` policy for that?
```suggestion
    """Tests that the cursor returns a timeout error in case of a timeout, when the timeout policy is
    `ON_TIMEOUT FAIL`"""
```